realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -196,6 +196,11 @@ public long addColumnLink (RealmFieldType type, String name, Table table) {
     @Override
     public void removeColumn(long columnIndex) {
         nativeRemoveColumn(nativePtr, columnIndex);
+        // When PK does not exist this cannot happen. But if you remove a column with a smaller index
+        // than that of PK column, you need to rearrange PK column index.
+        if (hasPrimaryKey() && columnIndex < getPrimaryKey()) {
```

Comment:
`columnIndex < getPrimaryKey()` -> ``columnIndex <= getPrimaryKey()`? If the column is the primary key, the cached index should be cleared.